### PR TITLE
Documentation: DevGuide - transactions

### PR DESCRIPTION
**Pull request by jufajardini**
- extending/applayer/transactions: add file with first draft

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
-- there isn't one

Describe changes:
- Summary of what I understood from Transactions with Victor
- Add some code snippets to help with visualization
- Add a glossary with common abbreviations that may not be so transparent to beginners 

Questions & additional clarifications: 
- Where should this file go? Is doc/devguide/extending/applayer/transactions the right place?
- I think I need more clarification regarding, especially:
    - The part about partial rule matching and protocols with large (PDUs?), felt too vague when I was trying to explain it
    - the abstraction for file transfer protocols. Not confident about what I wrote.
- Devs themselves will do the state handling for incrementing it, right?

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

File: doc/devguide/extending/app-layer/transactions.rst [hunk 1]
```diff
@@ -0,0 +1,112 @@
+************
+Transactions
+************
+
+Transactions are what guides when detecting and logging happen in Suricata. They’ll also help when dealing with protocols that can have large PDUs, like TCP, in controlling state for partial rule matching, for rules that mention more than one field.
```
Comment (victorjulien) [inline doc/devguide/extending/app-layer/transactions.rst:5]: "Transactions are abstractions that help guide detection and loggin in Suricata"

File: doc/devguide/extending/app-layer/transactions.rst [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
```
Comment (victorjulien) [inline doc/devguide/extending/app-layer/transactions.rst:6]: add something like: "transactions are implemented per parser and are stored in the per flow state. The engine interacts with it using a set of callbacks the parser registers."

File: doc/devguide/extending/app-layer/transactions.rst [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                        }
+                    }
+                }
+        .
+        .
+        .
+    }
+
+
+As a rule, transactions will follow a request-response model, meaning that if a transaction has had a request and a response, it is complete.
```
Comment (victorjulien) [inline doc/devguide/extending/app-layer/transactions.rst:45]: "rule of thumb"

**Review by victorjulien — CHANGES_REQUESTED** [review]
I think it would be useful to have some sub headers:
- general concepts
- how engine uses transactions
- progress tracking

**Comment by jufajardini** [comment]
Closed with https://github.com/OISF/suricata/pull/5974

**Pull request closed without merging**